util/fs/fs_test.go

```diff
@@ -82,48 +94,54 @@ func TestFSDetectorResize(t *testing.T) {
 
 			// Unformatted case.
 			if fsType, err := Detect(fakeDevicePath); err != ErrNoFilesystemDetected {
-				t.Fatalf("Expected ErrNoFilesystemDetected from Detect for unformatted device, but got fsType=%s err=%+v", fsType, err)
+				t.Fatalf("Expected ErrNoFilesystemDetected from Detect() for unformatted device, but got fsType=%v err=%+v", fsType, err)
 			}
 
 			// Unformatted case.
 			if err := Resize(fakeDevicePath); err == nil {
-				t.Fatal("Expected error from Resize for unformatted device, but got nothing")
+				t.Fatal("Expected error from Resize() for unformatted device, but got nothing; fsType=%v", targetFSType)
 			}
 
 			// Format it.
-			if err := FormatDevice(fakeDevicePath, targetFsType); err != nil {
-				t.Fatalf("Unexpected error formatting device=%s: %s", fakeDevicePath, err)
+			if err := FormatDevice(fakeDevicePath, targetFSType); err != nil {
+				t.Fatalf("Unexpected error formatting device=%v: %s", fakeDevicePath, err)
 			}
 
 			// Formatted case.
 			{
 				fsType, err := Detect(fakeDevicePath)
 				if err != nil {
-					t.Errorf("Unexpected error from Detect: %s", err)
+					t.Errorf("Unexpected error from Detect() for fsType=%v: %s", targetFSType, err)
 				}
-				if expected, actual := targetFsType, fsType; actual != expected {
-					t.Errorf("Wrong result from Detect, expected fsType=%q but actual=%q", expected, actual)
+				if expected, actual := targetFSType, fsType; actual != expected {
+					t.Errorf("Incorrect result from Detect(), expected fsType=%q but actual=%q", expected, actual)
 				}
 
 				err = Resize(fakeDevicePath)
 				if err != nil {
-					t.Errorf("Unexpected error in resize: %s", err)
+					t.Errorf("Unexpected error from Resize() for fsType=%v: %s", targetFSType, err)
 				}
 			}
 		}()
 	}
 }
 
-
 func makeFakeDevice(path string) error {
 	size := 100 // NB: btrfs requires a minimum size of 100MB.
-	if output, err := exec.Command("dd", "if=/dev/zero", fmt.Sprintf("of=%s", path), "bs=1M", fmt.Sprintf("count=%v", size)).CombinedOutput(); err != nil {
-		return fmt.Errorf("fake device setup failed: dd command output=%s err=%s", string(bytes.Trim(output, "\r\n \t")), err)
+	if output, err := exec.Command("dd", "if=/dev/zero", fmt.Sprintf("of=%v", path), "bs=1M", fmt.Sprintf("count=%v", size)).CombinedOutput(); err != nil {
+		return fmt.Errorf("fake device setup failed: dd command output=%v err=%s", string(bytes.Trim(output, "\r\n \t")), err)
 	}
 	return nil
 }
 
-func hasPasswordlessSudo() bool {
+func hasRoot() bool {
+	u, err := user.Current()
+	if err != nil {
+		panic(err)
+	}
+	if u.Uid == "0" {
```

Comment:
```bash
jtaylor@ubuntu:~/go/src/github.com/medallia/convoy$ go test -v ./util/fs/... ; echo $?
=== RUN   TestDeviceFormatter
--- PASS: TestDeviceFormatter (1.03s)
=== RUN   TestFSDetectorResize
--- PASS: TestFSDetectorResize (0.64s)
	fs_test.go:76: Skipping resize test for fsType=btrfs because resize2fs only supports ext2/ext3/ext4
	fs_test.go:76: Skipping resize test for fsType=minix because resize2fs only supports ext2/ext3/ext4
	fs_test.go:76: Skipping resize test for fsType=xfs because resize2fs only supports ext2/ext3/ext4
PASS
ok  	github.com/medallia/convoy/util/fs	1.675s
0
```

```bash
testaccount@ubuntu:/home/jtaylor/go/src/github.com/medallia/convoy$ /usr/local/go/bin/go test -v ./util/fs/... ; echo $?
=== RUN   TestDeviceFormatter
--- SKIP: TestDeviceFormatter (0.00s)
	fs_test.go:20: TestDeviceFormatter skipped because sudo privileges are not currently available
=== RUN   TestFSDetectorResize
--- SKIP: TestFSDetectorResize (0.00s)
	fs_test.go:61: TestFSDetectorResize skipped because sudo privileges are not currently available
PASS
ok  	_/home/jtaylor/go/src/github.com/medallia/convoy/util/fs	0.009s
0```